web/src/components/storage/ProposalPage.tsx

```diff
@@ -21,24 +21,215 @@
  */
 
 import React from "react";
-import { Content, Grid, GridItem, SplitItem } from "@patternfly/react-core";
-import { Page } from "~/components/core/";
-import { Loading } from "~/components/layout";
-import EncryptionField from "~/components/storage/EncryptionField";
+import {
+  Alert,
+  Button,
+  Content,
+  Grid,
+  GridItem,
+  Split,
+  SplitItem,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateFooter,
+  List,
+  ListItem,
+} from "@patternfly/react-core";
+import { Page, Link } from "~/components/core/";
+import { Icon, Loading } from "~/components/layout";
 import ProposalResultSection from "./ProposalResultSection";
 import ProposalTransactionalInfo from "./ProposalTransactionalInfo";
 import ProposalFailedInfo from "./ProposalFailedInfo";
 import ConfigEditor from "./ConfigEditor";
 import ConfigEditorMenu from "./ConfigEditorMenu";
 import AddExistingDeviceMenu from "./AddExistingDeviceMenu";
-import { toValidationError } from "~/utils";
+import { Issue, IssueSeverity, IssueSource } from "~/types/issues";
+import {
+  useAvailableDevices,
+  useConfigMutation,
+  useDeprecated,
+  useDeprecatedChanges,
+  useReprobeMutation,
+} from "~/queries/storage";
+import { useConfigModel } from "~/queries/storage/config-model";
+import { useZFCPSupported } from "~/queries/storage/zfcp";
+import { useDASDSupported } from "~/queries/storage/dasd";
 import { useIssues } from "~/queries/issues";
-import { IssueSeverity } from "~/types/issues";
-import { useDeprecated, useDeprecatedChanges, useReprobeMutation } from "~/queries/storage";
+import { STORAGE as PATHS } from "~/routes/paths";
 import { _ } from "~/i18n";
 
-export default function ProposalPage() {
+/** @todo Call to an API method to reset the default config instead of setting a config. */
+function useResetConfig() {
+  const { mutate } = useConfigMutation();
+
+  return () =>
+    mutate({
+      drives: [
+        {
+          partitions: [{ search: "*", delete: true }, { generate: "default" }],
+        },
+      ],
+    });
+}
+
+type ErrorsEmptyStateProps = {
+  errors: Issue[];
+};
+
+function ErrorsEmptyState({ errors }: ErrorsEmptyStateProps) {
```

Comment:
I think we have gone too far accumulating functionality on this file (just look at the size of the "import" section). Can't we extract the empty pattern component(s) to separate files? Similar to what is done for `ProposalTransactionalInfo` and `ProposalFailedInfo`?

Reply:
I don't know if extracting single-usage-components like these, which don't have any kind of logic, is good or not.

IMHO, in cases like this, I prefer to have such components defined directly there in the only file where they are needed. Otherwise, there is a lot chance of forgetting about them when the main component is refactored (that has already happened). Of course, there could be exceptions, like having complex components with too much logic.

But I can extract them if you guys prefer to go with that approach.

Reply:
We decided to keep that components directly there, but in the future we will extract that components to separate files. Each component will have the logic to decide whether to render any content. The proposal page will simply mount all those components. Moreover, each components will be tested separated.

Reply:
I didn't expect that after our conversation 2 or 3 hours ago. Why did you decide to postpone such an agreed movement?

Reply:
In a few words: "time pressure" 😓

Reply:
> Why did you guys decide to postpone something agreed less than 3 hours ago? Just curious because it is, again, a kick to the front-end code.

JFTR: because just after talking with you, the PO decided to go with the current code if it works. As @imobachgs said, time pressure. We agreed a) to add a FIXME, b) follow the agreed approach from now on, c) redo this code (and many other things) once time permits.

> I start having the feeling that does not payoff to invest time in meetings.

I disagree. We agreement is there and we will follow it from now on.